src/integTest/java/com/google/cloud/tools/gradle/appengine/AppEngineStandardPluginIntegrationTest.java

```diff
@@ -77,25 +78,41 @@ public void testDevAppServer_sync() {
    */
   @Test
   public void testDevAppServer_async() throws InterruptedException, IOException {
-    GradleRunner.create()
-        .withProjectDir(testProjectDir.getRoot())
-        .withPluginClasspath()
-        .withArguments("appengineStart")
-        .build();
-
-    AssertConnection.assertResponse(
-        "http://localhost:8080", 200, "Hello from the App Engine Standard project.");
-
-    GradleRunner.create()
-        .withProjectDir(testProjectDir.getRoot())
-        .withPluginClasspath()
-        .withArguments("appengineStop")
-        .build();
-
-    // give the server a couple seconds to come down
-    Thread.sleep(8000);
-
-    AssertConnection.assertUnreachable("http://localhost:8080");
+    try {
+      GradleRunner.create()
+          .withProjectDir(testProjectDir.getRoot())
+          .withPluginClasspath()
+          .withArguments("appengineStart")
+          .build();
+
+      File expectedLogFileDir = new File(testProjectDir.getRoot(), "/build/tmp/appengineStart");
+      DirectoryScanner ds = new DirectoryScanner();
+      ds.setIncludes(new String[] {"dev_appserver*.out"});
+      ds.setBasedir(expectedLogFileDir);
+      ds.scan();
+      String[] devAppserverLogFiles = ds.getIncludedFiles();
+      Assert.assertEquals(1, devAppserverLogFiles.length);
+      String devAppServerOutput =
+          FileUtils.readFileToString(new File(expectedLogFileDir, devAppserverLogFiles[0]));
+      System.out.println(devAppServerOutput);
+      Assert.assertTrue(
+          devAppServerOutput.contains("Dev App Server is now running")
+              || devAppServerOutput.contains("INFO:oejs.Server:main: Started"));
+
+      AssertConnection.assertResponse(
+          "http://localhost:8080", 200, "Hello from the App Engine Standard project.");
+    } finally {
+      GradleRunner.create()
+          .withProjectDir(testProjectDir.getRoot())
+          .withPluginClasspath()
+          .withArguments("appengineStop")
+          .build();
+
+      // give the server a couple seconds to come down
+      Thread.sleep(8000);
+
+      AssertConnection.assertUnreachable("http://localhost:8080");
```

Comment:
I took out this finally, it's not helpful when the first part of the test fails and then this fails, since the first exception is lost.